### PR TITLE
Create CONTRIBUTING.md

### LINKED ISSUE #909: [Feat]: Add CONTRIBUTING
### 背景与动机

https://github.com/DGP-Studio/Snap.Hutao/community

### 想要实现或优化的功能

- CONTRIBUTING should include guidance on
  - commit & PR
  - issue
  - document

### PR DESCRIPTION
**Pull request by Masterain98**
resolve #909

### CODE REVIEW

**Comment by Masterain98** [comment]
@Lightczx check and refine the `Contribute Your Code` part

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Contribution Guide
+
+## Contribute Your Code
+
+### Setup Snap.Hutao Project
+
+1. Download and install [Visual Studio 2022 Community](https://visualstudio.microsoft.com/downloads/)
+2. Open Visual Studio Installer to complete Visual Studio installation
+   - You need to install `.NET desktop development`, `Desktop development with C++` and `Universal Windows Platform development` components.
```
Comment (Lightczx) [inline CONTRIBUTING.md:9]: Single-project MSIX packaging Tools is required
image

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+## Contribute Your Code
+
+### Setup Snap.Hutao Project
+
+1. Download and install [Visual Studio 2022 Community](https://visualstudio.microsoft.com/downloads/)
+2. Open Visual Studio Installer to complete Visual Studio installation
+   - You need to install `.NET desktop development`, `Desktop development with C++` and `Universal Windows Platform development` components. 
+3. Use git to clone the project `https://github.com/DGP-Studio/Snap.Hutao.git` to your local device 
+   - You may also use our GitLab mirror repository `https://jihulab.com/DGP-Studio/Snap.Hutao.git` if you have trouble connecting to GitHub
```
Comment (Lightczx) [inline CONTRIBUTING.md:11]: Never recommand, no PR will be accept in jihu mirror

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+4. Switch git branch to `develop`
+5. Open project solution with your Visual Studio and then you are ready to go
+
+### Start Pull Request
+
+- All code-related changes should only be push to `develop` branch
```
Comment (Lightczx) [inline CONTRIBUTING.md:17]: to users any own branch, but only allow pull request target to develop branch.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+- [FAQ](https://hut.ao/advanced/FAQ.html) Document
+- [Common Program Exceptions ](https://hut.ao/en/advanced/exceptions.html)Document
+- [Current Opened BUG Report Issues](https://github.com/DGP-Studio/Snap.Hutao/issues?q=is%3Aissue+is%3Aopen+label%3ABUG)
+
+When starting a new issue, please use the issue templates:
+
+- If your issue is about program crash
+  - Remember to provide your Device ID
+  - Check Windows Event Viewer, and attach associated `.NET Error` details in the issue body
+- Describe your issue in details to help developers to reproduce the issue
```
Comment (Lightczx) [inline CONTRIBUTING.md:43]: repro steps please

File: CONTRIBUTING.md (hunk 1 above)
Comment (Masterain98) [reply CONTRIBUTING.md]: added

**Review by Lightczx — APPROVED** [review]
LGTM

**Pull request merged**